app/controllers/comments_controller.rb

```diff
@@ -1,20 +1,37 @@
 class CommentsController < ApplicationController
+  load_and_authorize_resource
   def new
     @comment = Comment.new
     @post = Post.find(params[:post_id])
   end
 
   def create
-    @post = Post.includes(:comments).find(params[:post_id]) # Eager loading comments for the specific post
+    @user = current_user
+    # Eager loading comments for the specific post
+    @post = Post.includes(:comments).find(params[:post_id])
     @comment = @post.comments.new(comment_params)
-    @comment.author = current_user
+    @comment.post = @post
+    @comment.author = @user
     if @comment.save
-      redirect_to user_post_path(current_user, @post)
+      redirect_to user_post_path(@user, @post)
     else
       render :new, status: :unprocessable_entity
     end
   end
 
+  def destroy
+    @comment = Comment.includes(:author, :post).find(params[:id])
+    user = @comment.author_id
+    post = @comment.post_id
+
+    if @comment.destroy
+      flash[:success] = 'Comment was successfully deleted.'
+    else
+      flash[:error] = 'Error: Comment could not be deleted.'
+    end
+    redirect_to user_post_path(user, post)
+  end
```

Comment:
- [ ] In order to be able to destroy a comment, it needs to be created first and I was unable to create a comment for a user's post when posts are created. Kindly make it possible for comments to be created so the destroy feature can work as expected 👍🏽

I keep getting this error on the screenshot below when I try to create a comment for a post. 

<img width="1208" alt="Screenshot 2023-07-29 at 1 26 22 PM" src="https://github.com/DJ-MrJay/My-Blog-App/assets/50941074/af55cadd-72fd-4df5-a12d-7718585f097c">

Reply:
Kindly call me on Zoom. Everything seems to be working on my end.